### PR TITLE
Some operators where not visible in node searcher

### PR DESCRIPTION
**Pull request by julienamsellem**
### Purpose of this PR
https://fogbugz.unity3d.com/f/cases/1390149/

Some operators where missing from the node searcher when dragging a connection from a port in a context.

Steps to reproduce
- Create a default VFX Graph and open it
- In the Output Particle Quad context, drag from the color port in the "Set Color over Life " block
- You should see Operator=>Inline=>Gradient
=> But the operator is missing

[Before]
gh8BGeJgDT

[After]
6rMFzcdREF

---
### Testing status
Tested with `Set Color over Life` and `Set Size over Life`

### CODE REVIEW

**Comment by VitaSkruibyte** [comment]
For testing used AllBlocksNodes.vfx
image

Testing dragging form block or node and and looked for a matching operators/ confirmed there are no empty search:
1. Checked all possible types (gradient, animation curve, camera, shape, space types, texture types, camera)
2. Blocks with various Compositions (Blend, Overwrite, Add, Multiply)
3. Also observed that empty search was specifically present with:
_- Angle blocks_
_- Shape blocks_
_- Bounds (IIRC)_
_- Attribute Map_
_- Signed Distance Field_
These now look good, Search display possible operators
4. Checked dragging from contexts to make sure things are good
5. Tried changing type lets say from curve to vector and then dragging an edge. Search display matching type of the operator

**Review by VitaSkruibyte — APPROVED** [review]

**Review by MarieGuff — APPROVED** [review]
All good
tested on 2022.2.0a1_2dd09014126d

**Review by PaulDemeulenaere — COMMENTED** [review]
LGTM to me, I've a just a question about a potential unexpected change.
Tested locally as well.
_tested_missing_inline

File: com.unity.visualeffectgraph/Editor/GraphView/Elements/VFXDataAnchor.cs [hunk 1]
```diff
@@ -361,16 +361,22 @@ bool ProviderFilter(VFXNodeProvider.Descriptor d)
...
                 if (op != null)
-                    validTypes = op.validTypes;
+                    validTypes = op.validTypes.ToArray();
             }
 
-            var getSlots = direction == Direction.Input ? (System.Func<int, VFXSlot>)container.GetOutputSlot : (System.Func<int, VFXSlot>)container.GetInputSlot;
+            var getSlots = direction == Direction.Input ? container.GetOutputSlot : (System.Func<int, VFXSlot>)container.GetInputSlot;
             int count = direction == Direction.Input ? container.GetNbOutputSlots() : container.GetNbInputSlots();
+            if (count == 0) // Template containers are not sync initially to save time during loading
+            {
```
[the remark this answers is not in the document]
Comment (julienamsellem) [reply com.unity.visualeffectgraph/Editor/GraphView/Elements/VFXDataAnchor.cs:375]: Yes for containers with no input or output, SyncSlots will be called every time. But today there are very few or even none in that case.
I'm adding a comment to stress on if we have a performance issue in the future.

**Review by PaulDemeulenaere — APPROVED** [review]
Double checked applied changed at https://github.com/Unity-Technologies/Graphics/pull/6662/commits/9d45d9db23cceb3086594c042524e1c1036cf33f
Thanks 🟢

**Pull request merged**